CIP-????/README.md

```diff
@@ -0,0 +1,294 @@
+
+```
+cardano.{walletName}.signAlgo
+cardano.{walletName}.hashAlgo
+```
+
+There should be additional extension registered if [CIP on wallet extensibility](https://github.com/cardano-foundation/CIPs/pull/462).
+
+Wallet providers may choose to automatically generate ultra-secure wallets for multiple quantum signature
+algorithms from a single seed. In such case, they are encouraged to provide a visible
+hint of what is the current preferred.
+
+### Use of standard hash algorithm
+
+Note that use of hash would require use of quantum-resistant hash algorithm.
+[SHA2, SHA3, BLAKE2 are considered quantum-safe](https://cryptobook.nakov.com/quantum-safe-cryptography#quantum-safe-and-quantum-broken-crypto-algorithms),
+in particular standard 256-bit hash used in Cardano [should be safe for a long time](https://cryptobook.nakov.com/quantum-safe-cryptography#quantum-safe-and-quantum-broken-crypto-algorithms).
+This should be documented in the Plutus documentation of `verifySignature`.
+
+However, in the interest of long-term API evolution we propose to add a flexible hashing interface:
+
+```
+data SignAlgo = SHA2_256
+              | SHA3_256
+              | Blake2
+
+hashWithAlgo :: HashAlgo -> BuiltinByteString -> BuiltinByteString
+```
+
+This will also allow for gradual _opt-in_ migration of scripts as the blockchain evolves.
+
+### Long term crypto API evolution
+
+It would be desirable to also offer a long-term API that would not need any changes
+as we migrate blockchain to new algorithms.
+
+This would be to provide an implicit embedding of a most secure available hashing algorithm
+within the hash itself, and use of interface:
+
+```
+data LTSignature = LongTermSignature {
+                     ltsHashAlgo :: HashAlgo
+                   , ltsSignAlgo :: SignAlgo
+                   , ltsSignature :: BuiltinByteString
+                   }
+
+verifyLongTermSignature :: BuiltinByteString                       -- ^ public key
+                        -> LongTermSignature                       -- ^ message or hash
+                        -> BuiltinByteString                       -- ^ signature
+                        -> Bool  -- is hash valid
+
+isLongTermSignatureSecure :: LongTermSignature -> Bool
+```
+
+Supporting evolution of hash algorithms is much simpler:
+
+```
+-- | Long term hash
+data LongTermHash = LongTermHash {
+                      hashAlgo :: HashAlgo
+                    , hash     :: BuiltingByteString
+                    }
+
+mostSecureHashAlgo :: HashAlgo
+
+mostSecureHash :: BuiltinByteString -> LongTermHash
+mostSecureHash = hashWithAlgo mostSecureHash
+```
+
+This API needs additional support from the off-chain applications in order to:
+
+* provide correct public key for the given algorithm
+* help users to transfer their euTxOs to new signature algorithms when old algorithms are obsoleted.
+
+Note that `LongTermHash` and `LongTermSignature` can be implemented as standard library functions.
```

Comment:
So this is just a proposal to add some stuff to the `plutus` standard library? That doesn't need a CIP! I suggest removing it to keep things slim and opening an issue in `plutus`.